### PR TITLE
MEN-5930: index device deployment objects

### PR DESCRIPTION
**Pull request by tranchitella**

### CODE REVIEW

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline --pr reporting/87

File: app/app.go [hunk 1]
```diff
@@ -1264,47 +1264,16 @@ func (d *Deployments) isDevicePartOfDeployment(
...
-			}
-			return nil, ErrConflictingRequestData
-		}
-	} else {
-		// save the request
-		if err := d.db.SaveDeviceDeploymentRequest(
-			ctx, deviceDeployment.Id, request); err != nil {
-			return nil, err
-		}
+	err = d.saveDeviceDeploymentRquest(ctx, deviceID, deviceDeployment, request)
```
Comment (alfrunes) [inline app/app.go:1274]:
```suggestion
	err = d.saveDeviceDeploymentRequest(ctx, deviceID, deviceDeployment, request)
```

File: app/app.go [hunk 2]
```diff
@@ -1394,6 +1364,45 @@ func (d *Deployments) GetDeploymentForDeviceWithCurrent(ctx context.Context, dev
...
+			// mark deployment for this device as failed to force client to rollback
+			l := log.FromContext(ctx)
+			l.Errorf(
+				"Device with id %s reported new data: %s during update process;"+
+					"old data: %s",
+				deviceID, request, deviceDeployment.Request)
+
+			if err := d.UpdateDeviceDeploymentStatus(ctx, deviceDeployment.DeploymentId, deviceID,
+				model.DeviceDeploymentState{
+					Status: model.DeviceDeploymentStatusFailure,
```
Comment (alfrunes) [inline app/app.go:1382]: Random observation: I feel like we should set the SubState here to make it obvious from the UI what happened.

File: app/app.go [hunk 3]
```diff
@@ -1741,11 +1765,16 @@ func (d *Deployments) updateDeviceDeploymentsStatus(
...
+				deviceDeployment, err := d.createDeviceDeploymentWithStatus(ctx,
 					deviceId, deployment, status)
 				if err != nil {
 					return err
 				}
+				if err := d.reindexDeployment(ctx, deviceDeployment.DeviceId,
+					deviceDeployment.DeploymentId, deviceDeployment.Id); err != nil {
+					l := log.FromContext(ctx)
+					l.Warn(errors.Wrap(err, "failed to trigger a device reindex"))
+				}
```
Comment (alfrunes) [inline app/app.go:1786]: Shouldn't there be a condition on the status being a terminal state (`success`, `failure`, `already-installed`, `noartifact`, `decommissioned` or `aborted`)?

File: app/app.go (hunk 2 above)
Comment (tranchitella) [reply app/app.go:1382]: The idea is valuable, but I don't think it fits into my PR.

File: app/app.go (hunk 3 above)
Comment (tranchitella) [reply app/app.go:1786]: you are right

**Pull request merged**